generated/nitclk/session.py

```diff
@@ -0,0 +1,1156 @@
+        nitclk.Session repeated capabilities container, and calling this method on the result.:
+
+            session.channels[0,1]._set_attribute_vi_session(attribute_id)
+
+        Args:
+            attribute_id (int): The ID of the property that you want to set Supported Properties
+                start_trigger_master_session
+                ref_trigger_master_session
+                script_trigger_master_session
+                pause_trigger_master_session
+
+        '''
+        session_ctype = _visatype.ViSession(self._session)  # case S110
+        channel_name_ctype = ctypes.create_string_buffer(self._repeated_capability.encode(self._encoding))  # case C010
+        attribute_id_ctype = _visatype.ViAttr(attribute_id)  # case S150
+        value_ctype = _visatype.ViSession(self._value)  # case S110
+        error_code = self._library.niTClk_SetAttributeViSession(session_ctype, channel_name_ctype, attribute_id_ctype, value_ctype)
+        errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
+        return
+
+    def _set_attribute_vi_string(self, attribute_id, value):
+        r'''_set_attribute_vi_string
+
+        Sets the value of an NI-TClk VIString property.
+        _set_attribute_vi_string is a low-level method that you can use to
+        set the values of NI-TClk properties. NI-TClk contain high-level
+        methods that set most of the properties. It is best to use the
+        high-level methods as much as possible.
+
+        Tip:
+        This method requires repeated capabilities (channels). If called directly on the
+        nitclk.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        nitclk.Session repeated capabilities container, and calling this method on the result.:
+
+            session.channels[0,1]._set_attribute_vi_string(attribute_id, value)
+
+        Args:
+            attribute_id (int): Pass the ID of the property that you want to set Supported Properties
+                sync_pulse_source
+                sync_pulse_clock_source
+                exported_sync_pulse_output_terminal
+
+            value (str): Pass the value for the property
+
+        '''
+        session_ctype = _visatype.ViSession(self._session)  # case S110
+        channel_name_ctype = ctypes.create_string_buffer(self._repeated_capability.encode(self._encoding))  # case C010
+        attribute_id_ctype = _visatype.ViAttr(attribute_id)  # case S150
+        value_ctype = ctypes.create_string_buffer(value.encode(self._encoding))  # case C020
+        error_code = self._library.niTClk_SetAttributeViString(session_ctype, channel_name_ctype, attribute_id_ctype, value_ctype)
+        errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
+        return
+
+
+class _Session(object):
+    '''Hidden class for template reuse'''
```

Comment:
I don't think docstrings in generated code should reference code-generation aspects. We should document what the class does in the context of the code it is documenting.

Reply:
Made longer to explain what is going on. The purpose of this class _is_ code generation related though so the comment is still about it, just more wordy.